app/facade.py

```diff
@@ -0,0 +1,259 @@
+async def create_controller(req_header: Header):
+    """
+    Instantiate an AriesAgentController or a TenantController
+    based on header attributes
+
+    Parameters:
+    -----------
+    req_header: Header
+        The header object containing (wallet_id, jwt_token) or api_key
+
+    Returns:
+    --------
+    controller: aries_cloudcontroller instance
+        A generator of AriesAgentController or TenantController object
+        (for use in contextmanager)
+    """
+    is_valid_tenant_header = "wallet_id" in req_header and "tenant_jwt" in req_header
+    is_valid_admin_header = "api_key" in req_header
+    is_valid_header = req_header and (is_valid_tenant_header or is_valid_admin_header)
+    if is_valid_header:
+        req_header = eval(req_header)
+        if "api_key" in req_header:
+            controller = AriesAgentController(
+                admin_url=f"{admin_url}:{admin_port}",
+                api_key=req_header["api_key"],
+                is_multitenant=is_multitenant,
+            )
+        else:
+            controller = AriesTenantController(
+                admin_url=f"{admin_url}:{admin_port}",
+                wallet_id=req_header["wallet_id"],
+                tenant_jwt=req_header["tenant_jwt"],
+            )
+    else:
+        raise HTTPException(
+            status_code=400,
+            detail="Bad headers. Either provide an api_key or both wallet_id and tenant_jwt",
+        )
+    try:
+        yield controller
+    except Exception as e:
+        # We can only log this here and not raise an HTTPExeption as
+        # we are past the yield. See here: https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-with-yield/#dependencies-with-yield-and-httpexception
+        logger.error(f"{e!r}")
+    finally:
+        await controller.terminate()
+
+
+async def create_did(controller):
+    """
+    Creates a DID against the ledger using an AriesController
+
+    Parameters:
+    -----------
+    controller: AriesController
+        The aries_cloudcontroller object
+
+    Returns:
+    --------
+    generate_did_response: dict
+        The response object from generating a DID on the ledger
+    """
+    generate_did_res = await controller.wallet.create_did()
+    if not generate_did_res["result"]:
+        logger.error(f"Failed to create DID:\n{generate_did_res}")
+        raise HTTPException(
+            status_code=404,
+            detail=f"Something went wrong.\nCould not generate DID.\n{generate_did_res}",
+        )
+    return generate_did_res
+
+
+async def post_to_ledger(url, payload):
+    """
+    Post the did payload to the ledger
+
+    Parameters:
+    -----------
+    url: str
+        The url of the ledger to post to
+    payload: dict
+        The payload to be posted of the form:
+        {
+            "network": "stagingnet",
+            "did": did_object["did"],
+            "verkey": did_object["verkey"],
+            "paymentaddr": "somestring",
+        }
+
+    Returns:
```

Comment:
would be keen to introduce a _type_ to hold the values that are returned. Especially now that we have this facade we can surface to the dev who uses this what they can expect - don't think it'd be very much work to be honest